backup/barman-montagu

```diff
@@ -0,0 +1,290 @@
+        return (v, False)
+    except docker.errors.NotFound:
+        print("Creating volume '{}'".format(name))
+        v = docker_client.volumes.create(name)
+        return (v, True)
+
+
+def volume_remove(name, docker_client=docker.client.from_env()):
+    try:
+        v = docker_client.volumes.get(name)
+        print("Removing volume '{}'".format(v.name))
+        v.remove()
+        return True
+    except docker.errors.NotFound:
+        return False
+
+
+def container_remove(name, docker_client=docker.client.from_env()):
+    try:
+        x = docker_client.containers.get(name)
+        print("Removing container '{}'".format(name))
+        x.remove(force=True)
+        return True
+    except docker.errors.NotFound:
+        return False
+
+
+# See comments in montagu:
+# https://github.com/vimc/montagu/pull/87/files#diff-6927d77cf5038b5393885eb816eb9d7a
+def exec_safely(container, cmd, check=False, stream=True):
+    api = container.client.api
+    exec_id = api.exec_create(container.id, cmd)['Id']
+    out = api.exec_start(exec_id, stream=stream)
+    if stream:
+        ret = []
+        for line in out:
+            print(line.decode("UTF-8"), end="")
+            ret.append(line)
+        out = b''.join(ret)
+    dat = api.exec_inspect(exec_id)
+    dat['Output'] = out
+    if check and dat['ExitCode'] != 0:
+        msg = out.decode("UTF-8").strip()
+        raise Exception("Exec failed with message '{}'".format(msg))
+    return dat
+
+
+def resolve_host_ip(host):
+    if host == "localhost" or host == "127.0.0.1":
+        return localhost_ip()
+    try:
+        parsed = socket.inet_aton(host)
+        if len(parsed) != 4:
+            raise Exception("Expected 4 octets in host if given as IP")
+        return host
+    except socket.error:
+        return socket.gethostbyname(host)
+
+
+# NOTE: not terribly portable (probably linux only).  An alternative,
+# but far more cryptic, approach is to use:
+#
+#     with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+#         s.connect(('8.8.8.8', 1))
+#         return s.getsockname()[0]
+#
+# which is adapted from various stackoverflow answers.
+def localhost_ip():
+    res = subprocess.run(["hostname", "--ip-address"], check=True,
+                         stdout=subprocess.PIPE)
+    return res.stdout.decode("UTF-8").strip()
+
+
+def main():
+    args = docopt.docopt(__doc__)
+    if args['setup']:
+        setup(args['<host>'], args['--password-group'], args['--image-tag'],
+              args['--pull-image'], args['--configure-cron'],
+              not args['--no-clean-on-error'])
+    elif args['status']:
+        status()
+    elif args['barman']:
+        barman(args['<args>'])
+    elif args['recover']:
+        recover(args['<backup-id>'], args['--wipe-first'])
+    elif args['destroy']:
+        destroy()
+    elif args['wipe-recover']:
+        wipe_recover()
+    else:
+        raise Exception("(this should never happen)")
+
```

Comment:
Pycharm wants an extra blank line here

Reply:
Interesting - `pep8` (1.7.0) didn't flag that so I wonder if Pycharm is using a different linter.  (pep8 also doesn't mind if there is one more line there